src/Catalog/CommitCollector.cs

```diff
@@ -56,10 +57,25 @@ public CommitCollector(Uri index, Func<HttpMessageHandler> handlerFunc = null)
                         batch.CommitTimeStamp == lastBatch.CommitTimeStamp, 
                         cancellationToken);
 
-                    front.Value = batch.CommitTimeStamp;
-                    await front.Save(cancellationToken);
+                    // Only commit if this is the last batch that is going to be processed or if the commit timestamp
```

Comment:
I'm concered about putting this fix in the collectors base class.
It's only relevant for RegistrationCollector. Since the code base is not well UTed, and we dont have monitoring in place yet, this change introduces a risk. 
Lets discuss F2F.